src/response/file.rs

```diff
@@ -219,31 +219,31 @@ impl Stream for FileStream {
     type Item = Result<Bytes, DynError>;
 
     fn poll_next(self: Pin<&mut Self>, context: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let this = self.get_mut();
-
-        if this.remaining == 0 {
-            this.file = None;
-            this.buffer.fill(MaybeUninit::uninit());
+        if self.remaining == 0 {
             return Poll::Ready(None);
         }
 
+        let this = self.get_mut();
         let mut data = ReadBuf::uninit(&mut this.buffer);
-        let mut file = match this.file.as_mut() {
-            None => return Poll::Ready(None),
-            Some(file) => file,
-        };
 
-        match Pin::new(&mut file).poll_read(context, &mut data) {
+        match Pin::new(&mut this.file).poll_read(context, &mut data) {
             Poll::Pending => Poll::Pending,
 
             Poll::Ready(Ok(())) => {
                 let filled = data.filled();
-                this.remaining -= filled.len();
-                Poll::Ready(Some(Ok(Bytes::copy_from_slice(filled))))
+
+                if let Some(remaining) = this.remaining.checked_sub(filled.len()) {
+                    this.remaining = remaining;
```

Comment:
The code is sync here and not in a loop. It's easier to work directly with `this` as if it was `self` rather than storing `let remaining = &mut this.remaining`... I feel the same way about buffer. If we were in a loop, I'd treat things differently.